lib/Mojo/WebSocketProxy/Backend/ConsumerGroups.pm

```diff
@@ -336,13 +367,14 @@ sub _on_message {
         return;
     }
 
-    my $missed_param;
-    if(
-        ref $message ne 'HASH' 
-        || !$message->{$missed_param = "message_id"} 
-        || !$message->{$missed_param = "response"}
-    ) {
-        $log->errorf("Failed to process response: '%s' does not exist, original message content was %s", $missed_param, $raw_message);
+    if (ref $message ne 'HASH') {
+        $log->errorf("Failed to process response: Invalid message type got: %s, want: HASH", ref $message);
+        return;
+    }
+
+    my (@missing_params) = grep { !exists $message->{$_} } REQUIRED_RESPONSE_PARAMETERS;
+    if (scalar @missing_params) {
```

Comment:
No need for explicit `scalar`, without any other expressions the if conditional already implies scalar/boolean context:
```suggestion
    if (@missing_params) {
```

Reply:
But I don't think it is a bad habit.